src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1441,11 +1443,11 @@ uct_dc_mlx5_ep_arbiter_purge_internal_cb(ucs_arbiter_t *arbiter,
     }
 
     if (ucs_unlikely(req->func == uct_dc_mlx5_iface_fc_grant)) {
-        /* User callback should not be called for FC messages. Just return
-         * pending request memory to the pool */
-        freq = ucs_derived_of(req, uct_rc_pending_req_t);
-        ucs_mpool_put(freq);
-        return UCS_ARBITER_CB_RESULT_REMOVE_ELEM;
+        ucs_assertv(ep == iface->tx.fc_ep, "iface=%p: ep=%p fc_ep=%p", iface,
+                    ep, iface->tx.fc_ep);
+    } else {
+        ucs_assertv(ep != iface->tx.fc_ep, "iface=%p: ep=%p fc_ep=%p", iface,
+                    ep, iface->tx.fc_ep);
```

Comment:
it makes it more complicated. E.g. now we will handle fc grant requests with the path intended for non=intrnal requests

Reply:
fc_grant request could be scheduled on FC endpoint only.
so, I've changed behavior to not expect it on normal endpoints.
also, updated PR to remove internal_cb which is not needed anymore